fake/fake_clock_test.go

```diff
@@ -668,3 +668,90 @@ func TestFakeClockAfterFuncNegDur(t *testing.T) {
 	}
 
 }
+
+func TestFakeClockContext(t *testing.T) {
+	t.Run("ContextDeadline", func(t *testing.T) {
+		base := time.Now()
+		c := NewClock(base)
+
+		deadline := base.Add(1)
+		ctx, cancel := c.ContextWithDeadline(context.Background(), deadline)
+		t.Cleanup(cancel)
+
+		ctxDeadline, isSet := ctx.Deadline()
+		if !isSet {
+			t.Errorf("context deadline not set")
+		}
+		if !ctxDeadline.Equal(deadline) {
+			t.Errorf("unexpected context deadline: %v; expected %v", ctxDeadline, deadline)
+		}
+	})
+
+	t.Run("ContextWithDeadlineExceeded", func(t *testing.T) {
+		base := time.Now()
+		c := NewClock(base)
+
+		ctx, cancel := c.ContextWithDeadline(context.Background(), base.Add(1))
+		t.Cleanup(cancel)
+
+		c.Advance(1)
+
+		select {
+		case <-ctx.Done():
+			if ctx.Err() != context.DeadlineExceeded {
+				t.Errorf("unexpected error: %v; expected %v", ctx.Err(), context.DeadlineExceeded)
+			}
+		case <-time.After(time.Second):
+			t.Errorf("context not done after 1 second")
```

Comment:
You shouldn't need this.
Although the `AfterFunc` calls are run in separate goroutines, `Advance` and `SetClock` wait on a [waitgroup tracking those goroutines](https://github.com/vimeo/go-clocks/blob/ed66e788615d679769c2a0456b91501e31e0236a/fake/fake_clock.go#L26-L28).

https://github.com/vimeo/go-clocks/blob/ed66e788615d679769c2a0456b91501e31e0236a/fake/fake_clock.go#L122

Reply:
ok, so this would not be needed specifically due to verifying that Advance() returns 1?

Reply:
Correct